### ISSUE TITLE
Python: bindlink module misnamed

### ISSUE THREAD

**Issue by linas** [opencog/atomspace#2412]
The bindlink module is misnamed. So, for example, in ` tests/cython/atomspace/test_atom.py`  we see 
```
from opencog.bindlink import execute_atom
```
bindlink should be renamed `exec` so that
```
from opencog.exec import execute_atom
```
This is followup to pull req #2398

**Comment by linas** [comment]
@noskill @vsbogd fyi

**Comment by chtnnh** [comment]
Hello @linas , could I help fix this issue? I am new to OpenCog and wish to help out with open AGI. 

Are there any other files where this change needs to be made?

**Comment by linas** [comment]
@chtnnh 

> could I help fix this issue?

Sure!

> Are there any other files where this change needs to be made?

Possibly many, in other git repos.  Which ones is unknown. So the plan would be to make the change, then test. I think that almost certainly, we need a backwards-compat module called "bindlink" that prints a warning mesage, because I'm pretty sure that there are users who don't have github repos...

**Comment by chtnnh** [comment]
I can help change this particular instance and maybe even build the backwards-compat module "bindlink" to print out a warning message. Anything I need to know to get up and running? I understand this project is very complex and for very advanced programmers!

**Comment by linas** [comment]
> Anything I need to know to get up and running?

You should certainly be able to build, install and run the unit tests.  After that, go through at least some of the examples, to get some idea of what the system does.

> this project is very complex and for very advanced programmers!

It's complex in the same way that compilers are complex, or databases are complex.  You can certainly use a compiler, or a database, without understanding of what happens inside of it.  But if you want to actually change/improve/update the internals of a compiler/database, well, then, those are complex, and you need to understand a lot of comp-sci theory. As it happens, the AtomSpace borrows many ideas from compilers/databases, and I would like to see it borrow even more ...

**Comment by chtnnh** [comment]
> But if you want to actually change/improve/update the internals of a compiler/database, well, then, those are complex, and you need to understand a lot of comp-sci theory

I understand. Is there room for someone who is deeply passionate about the project and willing to learn the theory along the way? As a fresher in my undergraduate program I only have basic introduction to formal languages and compiler theory. But AGI is a passion of mine and opencog seems like an amazing project.

**Comment by linas** [comment]
> Is there room for someone who is ...

Yes! There's a large variety of things to do. Most of the bug reports don't require much theory, although many require understanding the internals fairly well ... and so this is like other coding projects, where you just have to spend a lot of time studying the code before being able to do anything with it.  Then there are various users of this project, who eithr need help in various ways, or are simply playgrounds for experimenting with new ideas. I'm mostly excited by my most recent, brand-new-project, https://github.com/opencog/generate ... its on version 0.0.5 so there is very little code to read and understand, although the theory might be a bit challenging.  I think its on the cutting edge. But this is probably ot quite the place for general discussion...

**Comment by chtnnh** [comment]
Yes I agree. Can I catch you on slack?

**Comment by linas** [comment]
IRC?

**Comment by linas** [comment]
For IRC, #opencog on freenode.  The problem with slack is lack of chatlogs and other nice features.

**Comment by linas** [comment]
Fixed in #2550

**Comment by ntoxeg** [comment]
You forgot to close.
